db/models.py

```diff
@@ -50,3 +54,70 @@
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        AUTH_USER_MODEL, on_delete=models.CASCADE, related_name="orders"
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    movie_session = models.ForeignKey(
+        to=MovieSession, on_delete=models.CASCADE, related_name="tickets"
+    )
+    order = models.ForeignKey(
+        to=Order, on_delete=models.CASCADE, related_name="tickets"
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["movie_session", "row", "seat"], name="unique_ticket"
+            )
+        ]
+
+    def clean(self) -> None:
+        cinema_hall = self.movie_session.cinema_hall
+
+        if self.row > cinema_hall.rows:
```

Comment:
The `clean` method should ensure that the row number is greater than or equal to 1. Consider adding a check for `self.row < 1` to raise a `ValidationError` if the row number is less than 1.

Reply:
Agree